classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,354 @@
+	 * translated before rendering
+	 */
+	protected $columns = array(
+		'hook'   => 'Hook',
+		'status' => 'Status',
+		'args'   => 'Arguments',
+		'group'  => 'Group',
+		'recurrence' => 'Recurrence',
+		'scheduled'  => 'Scheduled Date',
+		'claim_id'   => 'Claim ID',
+		'comments'   => 'Log',
+	);
+
+	protected $row_actions = array(
+		'hook' => array(
+			'run' => array( 'Run', 'Process the action now as if it were run as part of a queue' ),
+		),
+	);
+
+	/**
+	 *  The active data stores
+	 */
+	protected $store;
+
+	/**
+	 * Bulk actions. The key of the array is the method name of the implementation:
+	 *
+	 *     bulk_<key>(array $ids, string $sql_in).
+	 *
+	 * See the comments in the parent class for further details
+	 */
+	protected $bulk_actions = array(
+		'delete' => 'Delete',
+	);
+
+
+	/**
+	 * If it is true it will load our own javascript library.
+	 *
+	 * Our javascript library will show the logs as a jQuery modal.
+	 */
+	protected static $should_include_js = false;
+
+
+	/**
+	 * Sets the current data store object into `store->action` and initialises the object.
+	 */
+	public function __construct() {
+		self::$should_include_js = true;
+
+		$this->store = (object) array(
+			'action' => ActionScheduler_Store::instance(),
+			'log'    => ActionScheduler_Logger::instance(),
+		);
+
+		parent::__construct( array(
+			'singular' => $this->translate( 'action-scheduler' ),
+			'plural'   => $this->translate( 'action-scheduler' ),
+			'ajax'     => false,
+		) );
+	}
+
+	/**
+	 * Setups the libraries. It is executed if `is_admin()` is TRUE.
+	 */
+	public static function init() {
+		add_action( 'admin_footer', __CLASS__ . '::print_script' );
+		add_action( 'admin_enqueue_scripts', __CLASS__ . '::register_javascript' );
+	}
+
+	/**
+	 * Prints our javascript, if we need it.
+	 */
+	public static function print_script() {
+		if ( ! self::$should_include_js ) {
+			return;
+		}
+		wp_enqueue_script( 'action-scheduler' );
+		wp_print_styles( 'wp-jquery-ui-dialog' );
+	}
+
+	/**
+	 * Registers our javascript library and its dependencies.
+	 */
+	public static function register_javascript() {
+		wp_register_script( 'action-scheduler', plugins_url( 'action-scheduler.js', dirname( __FILE__ ) ), array(
+			'jquery',
+			'jquery-ui-core',
+			'jquery-ui-dialog',
+		), '1.0', true );
+	}
+
+	/**
+	 * Returns the recurrence of an action or 'Non-repeating'. The output is human readable.
+	 *
+	 * @param ActionScheduler_Action $item
+	 *
+	 * @return string
+	 */
+	protected function get_recurrence( $item ) {
+		$recurrence = $item->get_schedule();
+		if ( method_exists( $recurrence, 'interval_in_seconds' ) ) {
+			return self::human_interval( $recurrence->interval_in_seconds() );
+		}
+		return __( 'Non-repeating', 'action-scheduler' );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function get_items_query_limit() {
+		return $this->get_items_per_page( $this->package . '_items_per_page', $this->items_per_page );
+	}
```

Comment:
Would this method be better suited to the parent class?

Reply:
We need to override this. The parent class will return SQL code for the pagination, in our version we do not need SQL but the real number.